webui/src/lib/hooks/conf.jsx

```diff
@@ -0,0 +1,17 @@
+import React, {createContext, useContext} from "react";
+
+import {useAPI} from "./api";
+import {setup} from "../api";
+
+
+export const LoginConfigContext = createContext(null);
+
+export const WithLoginContext = ({children}) => {
+    const { response, error, loading } = useAPI(() => setup.getState());
+    const value = (error || loading) ? null : response?.login_config;
```

Comment:
`useAPI` promises a stable response. However, `value` isn't a stable reference.  
If you assume that the provider is at the top of the tree, it might not be a huge deal, but I'd still recommend `useMemo` or `useRef` for `value` to keep referential stability as it's a prop of the provider.

Reply:
Ouch, great catch --  thanks!